agent/grpc-external/services/peerstream/subscription_blocking.go

```diff
@@ -93,30 +93,27 @@ func (m *subscriptionManager) syncViaBlockingQuery(
 
 	store := m.getStore()
 
-	for {
+	for ctx.Err() == nil {
```

Comment:
Could you elaborate why `ctx.Err() == nil` is in the for condition? In the original version, the error msg will be caught by 
```go
} else if err != nil {
			logger.Error("failed to wait before re-trying sync", "error", err)
```
Now we seem to miss handling the error.